pkg/controller/hyperconverged/hyperconverged_controller_components_test.go

```diff
@@ -1164,69 +1167,361 @@ var _ = Describe("HyperConverged Components", func() {
+					Status:  corev1.ConditionTrue,
+					Reason:  "Foo",
+					Message: "Bar",
+				},
+				conditionsv1.Condition{
+					Type:    conditionsv1.ConditionProgressing,
+					Status:  corev1.ConditionFalse,
+					Reason:  "Foo",
+					Message: "Bar",
+				},
+				conditionsv1.Condition{
 					Type:    conditionsv1.ConditionDegraded,
+					Status:  corev1.ConditionFalse,
+					Reason:  "Foo",
+					Message: "Bar",
+				},
+			}
+
+			cl := initClient([]runtime.Object{hco, expectedResource})
+			r := initReconciler(cl)
+
+			// force upgrade mode
+			r.upgradeMode = true
+
+			res := r.ensureKubeVirtCommonTemplateBundle(req)
+
+			foundResource := &sspv1.KubevirtCommonTemplatesBundle{}
+			Expect(
+				cl.Get(context.TODO(),
+					types.NamespacedName{Name: expectedResource.Name, Namespace: expectedResource.Namespace},
+					foundResource),
+			).To(BeNil())
+
+			Expect(foundResource.ObjectMeta.Annotations).To(Not(HaveKey(hcoForceUpdate)))
+
+			Expect(res.UpgradeDone).To(BeFalse())
+			Expect(res.Updated).To(BeFalse())
+			Expect(res.Err).ToNot(HaveOccurred(), "forcing a reconciliation loop on SSP operator updating a label on KubevirtCommonTemplatesBundle")
+		})
+
+		It("should update an outdated annotation with a timestamp if KubeVirtCommonTemplateBundle is progressing and HCO is in upgrade mode", func() {
+			expectedResource := hco.NewKubeVirtCommonTemplateBundle()
+			expectedResource.ObjectMeta.SelfLink = fmt.Sprintf("/apis/v1/namespaces/%s/dummies/%s", expectedResource.Namespace, expectedResource.Name)
+			expectedResource.Status.Conditions = []conditionsv1.Condition{
+				conditionsv1.Condition{
+					Type:    conditionsv1.ConditionAvailable,
 					Status:  corev1.ConditionTrue,
-					Reason:  "KubevirtCommonTemplatesBundleDegraded",
-					Message: "KubevirtCommonTemplatesBundle is degraded: Bar",
-				})))
-			})
-		*/
+					Reason:  "Foo",
+					Message: "Bar",
+				},
+				conditionsv1.Condition{
+					Type:    conditionsv1.ConditionProgressing,
+					Status:  corev1.ConditionTrue,
+					Reason:  "Foo",
+					Message: "Bar",
+				},
+				conditionsv1.Condition{
+					Type:    conditionsv1.ConditionDegraded,
+					Status:  corev1.ConditionFalse,
+					Reason:  "Foo",
+					Message: "Bar",
+				},
+			}
+			var period int64 = 12 * 60
+			timestamp := time.Now().Unix()
+			ann := make(map[string]string)
+			ann[hcoForceUpdate] = fmt.Sprint((timestamp - 2*period) / period * period)
+			expectedResource.ObjectMeta.Annotations = ann
+
+			cl := initClient([]runtime.Object{hco, expectedResource})
+			r := initReconciler(cl)
+
+			// force upgrade mode
+			r.upgradeMode = true
+
+			res := r.ensureKubeVirtCommonTemplateBundle(req)
+
+			foundResource := &sspv1.KubevirtCommonTemplatesBundle{}
+			Expect(
+				cl.Get(context.TODO(),
+					types.NamespacedName{Name: expectedResource.Name, Namespace: expectedResource.Namespace},
+					foundResource),
+			).To(BeNil())
+
+			Expect(foundResource.ObjectMeta.Annotations).To(HaveKey("hcoForceUpdate"))
```

Comment:
@tiraboschi  Should not we check that the value is updated?

Reply:
The exact value is time dependent so we cannot really hard-code the expected value in a test (at least not easily without mocking time.Now()); checking against a variable unknown value is not that different that simply checking that at least we have a new annotation there.
Then we also have another test that explicitly checks if we are updating outdated values when needed.

Reply:
No sorry, you were right. I'll address this is an additional PR.

Reply:
Done, plese see:
https://github.com/kubevirt/hyperconverged-cluster-operator/pull/1256

Reply:
Thanks!